corehq/motech/value_source.py

```diff
@@ -104,6 +105,15 @@ def __eq__(self, other):
         )
 
     @classmethod
+    def get_schema_dict(cls) -> dict:
```

Comment:
What do you think of accepting `**options` here to facilitate subclasses constructing an augmented schema?

Reply:
I'm afraid I don't understand. Obviously subclasses are already adding their own properties to their schema, so I don't think that's what you mean. And currently this method is only used by `as_jsonobject()` ... so I think you mean that `as_jsonobject()` would need to accept `**options` to pass to `get_schema_dict()` ... but would (should?) callers of `as_jsonobject()` be able to determine whether the data they pass is valid or not? (Keeping in mind that the schema doesn't affect the definition of its class.)

If that is what you mean, then I'm worried that gives the caller too much rope. I would imagine that if the schema is rejecting valid values, that's a bug, and not up to the caller to resolve. ... The caller might want to modify the schema for it to transform the values it is passing ... but then why should the caller not transform the values before passing them?

I think I'm missing something.